src/textual/widget.py

```diff
@@ -2098,7 +2100,16 @@ def get_pseudo_classes(self) -> Iterable[str]:
             Names of the pseudo classes.
 
         """
-        yield "disabled" if self._self_or_ancestors_disabled else "enabled"
+        # NOTE: The heart of this yield is a direct copy of
+        # _self_or_ancestors_disabled. Because this method is called so
+        # much, here we save one function call as a very small but long-term
+        # useful optimisation. If _self_or_ancestors_disabled ever changes,
+        # be sure to reflect that change here!
+        yield "disabled" if any(
+            node.disabled
+            for node in self.ancestors_with_self
+            if isinstance(node, Widget)
+        ) else "enabled"
```

Comment:
@willmcgugan would something like `any(getattr(node, "disabled", False) for node in self.ancestors_with_self)` be potentially any faster? It's still one function call but now we don't use the dotted attribute access...
I'm trying to figure out these micro-optimisation heuristics.

Reply:
My gut reaction is that the attribute access makes more sense.

```py
import dis

def via_prop():
    return ( 1 ).__class__

def via_getattr():
    return getattr( 1, "__class__", "unknown" )

print( "via_prop" )
print( "=" * 40 )
dis.dis( via_prop )

print( "via_getattr" )
print( "=" * 40 )
dis.dis( via_getattr )
```

Reply:
@rodrigogiraoserrao `getattr` is a function itself, and a global which tends to be slow. Attribute is typically faster than globals.

@davep You can keep unrolling here. Unroll `ancestors_with_self`. There's some code a few lines down which does something similar.

Reply:
> You can keep unrolling here. Unroll ancestors_with_self. 

Done.